tests/end-to-end/verify_diff.py

```diff
@@ -0,0 +1,141 @@
+    for key, _ in result_model['state_dict'].items():
+        val = torch.sum((result_model['state_dict'][key] -
+                         expected_model['state_dict'][key]) ** 2).item()
+
+    if (val > 1e-3):
+        raise ValueError("Models not identical!")
+        sys.exit(-1)
+
+
+def _get_metrics_dict(logfile, strings_to_check):
+    """Parse a log file and look for specific strings.
+
+    Args:
+        logfile: file to be parsed.
+        strings_to_check: List of strings to look for and parse.
+
+    Return:
+        Dictionary with relevant strings as keys and the parsed
+        string as values.
+
+    """
+    metrics_dict = {}
+    with open(logfile, "r") as infile:
+        line = infile.readline()
+        while line:
+            for str_check in strings_to_check:
+                if (line.find(str_check) == 0):
+                    newstr = line.replace(str_check + " : ", '')
+                    metric_list = newstr.strip().split("|")
+                    metrics_dict[str_check] = metric_list
+            line = infile.readline()
+    return metrics_dict
+
+
+def verify_text(result_path, expected_path):
+    """Parse text file for particular metrics.
+
+    Compare with expected results.
+
+    Args:
+        result_path: Path to generated result file.
+        expected_path: Path to expected result file.
+
+    """
+    strings_to_check = ["Regression metrics on full data",
+                        "Regression metrics in peaks",
+                        "Regression metrics outside peaks",
+                        "Classification metrics at threshold 0.5",
+                        "AUC metrics"]
+
+    result_metrics = _get_metrics_dict(result_path, strings_to_check)
+    expected_metrics = _get_metrics_dict(expected_path, strings_to_check)
+
+    if (result_metrics.keys() != expected_metrics.keys()):
+        raise ValueError(
+            "Result file and expected file do not have matching metrics!")
+
+    for key in result_metrics.keys():
+        result_dict = {}
+        expected_dict = {}
+
+        for metric in result_metrics[key]:
+            metric = metric.strip().split(":")
+            result_dict[metric[0]] = float(metric[1])
+        for metric in expected_metrics[key]:
+            metric = metric.strip().split(":")
+            expected_dict[metric[0]] = float(metric[1])
+
+        if result_dict.keys() != expected_dict.keys():
+            raise ValueError(
+                "Metric keys for result and expected do not match!")
+
+        for key, value in result_dict.items():
+            diff = abs(value - expected_dict[key])
+            if diff > EPS:
+                err_msg = "Metrics for " + key + " differ more than " + str(
+                    EPS) + " between result and expected metrics!"
+                raise ValueError(err_msg)
+
+
+args = parse_args()
+if args.format == "torch":
+    verify_torch(args.result_path, args.expected_path)
+if args.format == "text":
```

Comment:
can a user pass both text and torch, or is it one or the other? if it's the latter it would be better to use `if... elif... else` instead of `if... if...`

Reply:
done.